solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -369,6 +378,77 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         else store.removePoolFromWhitelist(_poolAnchor);
     }
 
+    /**
+     * @dev adds a high tier pool
+     * can only be called by the contract owner
+     *
+     * @param _poolAnchor pool anchor
+     */
+    function addHighTierPool(IConverterAnchor _poolAnchor)
+        external
+        ownerOnly
+        validAddress(address(_poolAnchor))
+        notThis(address(_poolAnchor))
+    {
+        // validate input
+        PoolIndex storage poolIndex = highTierPoolIndices[_poolAnchor];
+        require(!poolIndex.isValid, "ERR_POOL_ALREADY_EXISTS");
+
+        poolIndex.value = _highTierPools.length;
+        _highTierPools.push(_poolAnchor);
+        poolIndex.isValid = true;
+    }
+
+    /**
+     * @dev removes a high tier pool
+     * can only be called by the contract owner
+     *
+     * @param _poolAnchor pool anchor
+     */
+    function removeHighTierPool(IConverterAnchor _poolAnchor)
+        external
+        ownerOnly
+        validAddress(address(_poolAnchor))
+        notThis(address(_poolAnchor))
+    {
+        // validate input
+        PoolIndex storage poolIndex = highTierPoolIndices[_poolAnchor];
+        require(poolIndex.isValid, "ERR_POOL_DOESNT_EXIST");
+
+        uint256 index = poolIndex.value;
+        uint256 length = _highTierPools.length;
+        assert(length > 0);
+
+        uint256 lastIndex = length - 1;
+        if (index < lastIndex) {
+            IConverterAnchor lastAnchor = _highTierPools[lastIndex];
+            highTierPoolIndices[lastAnchor].value = index;
+            _highTierPools[index] = lastAnchor;
+        }
+
+        _highTierPools.pop();
+        delete highTierPoolIndices[_poolAnchor];
+    }
+
+    /**
+     * @dev returns the list of high tier pools
+     *
+     * @return list of high tier pools
+     */
+    function highTierPools() external view returns (IConverterAnchor[] memory) {
+        return _highTierPools;
+    }
+
+    /**
+     * @dev checks whether a given pool is a high tier one
+     *
+     * @param _poolAnchor pool anchor
+     * @return true if the given pool is a high tier one, false otherwise
+     */
+    function isHighTierPool(IConverterAnchor _poolAnchor) public view returns (bool) {
+        return highTierPoolIndices[_poolAnchor].isValid;
+    }
+
```

Comment:
Missing:
```
    function highTierPoolCount() external view returns (uint256) {
        return _highTierPools;
    }
```

Reply:
yeah didn't want it there - you can get the array and check its length

Reply:
Makes me wonder if we even need all of those `Count` methods that we've added up until now in various places.
Perhaps it becomes useful once the array is extremely large, and a node might reject your attempt to read it in full.
@lbeder : maybe you have an idea?

Reply:
`*Count` methods are indeed very useful when the array is very large (not because it rejects, but to save undeeded resources using web3.js)

Reply:
yeah it's not relevant in this case, temp solution for very small number of pools.